mlir/lib/Transforms/Mem2Reg.cpp

```diff
@@ -472,9 +499,11 @@ LogicalResult mlir::tryToPromoteMemorySlots(
       if (slot.ptr.use_empty())
         continue;
 
-      SlotPromoter promoter(slot, allocator, builder, dominance);
-      if (succeeded(promoter.prepareSlotPromotion()))
-        toPromote.emplace_back(std::move(promoter));
+      SlotPromotionAnalyzer analyzer(slot, dominance);
+      Optional<SlotPromotionMetadata> metadata = analyzer.computeMetadata();
```

Comment:
metadata is maybe a bit overused for a project that sits in the llvm-project directory. 

SlotPromotionAnalysis, SlotPromotionInfo (I know terrible), I know you have a better idea :)